sample/sample.go

```diff
@@ -111,6 +111,7 @@ var samplerMetrics = []metrics.Metadata{
 	{Name: "_num_dropped", Type: metrics.Counter, Unit: metrics.Dimensionless, Description: "Number of traces dropped by configured sampler"},
 	{Name: "_num_kept", Type: metrics.Counter, Unit: metrics.Dimensionless, Description: "Number of traces kept by configured sampler"},
 	{Name: "_sample_rate", Type: metrics.Histogram, Unit: metrics.Dimensionless, Description: "Sample rate for traces"},
+	{Name: "_trace_key_length", Type: metrics.Histogram, Unit: metrics.Dimensionless, Description: "Number of unique trace keys"},
```

Comment:
I find this name confusing. At first I thought it was referring to the length of trace.trace_id. Could we rename it to "_sampler_key_cardinality", perhaps, and change the description to "Number of unique keys being tracked by the sampler"?